lib/credit_card_sanitizer.rb

```diff
@@ -170,7 +172,22 @@ def is_tracking?(candidate, options)
   end
 
   def valid_numbers?(candidate, options)
-    LuhnChecksum.valid?(candidate.numbers) && valid_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+    LuhnChecksum.valid?(candidate.numbers) && valid_company_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+  end
+
+  def valid_context?(options)
+    return true if valid_prefix?(options[:prefix]) && valid_postfix?(options[:postfix])
+    !options[:parse_flanking]
+  end
+
+  def valid_prefix?(prefix)
+    return true if prefix.nil? || prefix.downcase.end_with?(*ACCEPTED_PREFIX)
+    !!/[^[:alnum:]]/.match(prefix[-1])
```

Comment:
It turns out precompiling the `Regexp` doesn't help performance; Ruby must be doing some smart caching of regex expressions
So the set performance is hard to beat, but I would try to avoid `downcase` since the input string may be long and `downcase` will do a string allocation and go through the whole Unicode uppercase->lowercase rigamarole